### ISSUE TITLE
format_linux_int.c:297:25: error: ‘SIOCGSTAMP’ undeclared (first use in this function); did you mean ‘SIOCGARP’?

### ISSUE THREAD

**Issue by jer-gentoo** [LibtraceTeam/libtrace#117]
https://bugs.gentoo.org/692430

With Linux headers 5.2 and greater, format_linux_int.c fails to compile.
```
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -fvisibility=hidden -pthread -std=gnu99 -frecord-gcc-switches -g -pipe -O2 -Wall -march=amdfam10 -mtune=amdfam10 -Wno-comment -Wall -Wmissing-prototypes -Wextra -DLT_BUILDING_DLL=1 -c format_linux_int.c  -fPIC -DPIC -o .libs/format_linux_int.o
format_linux_int.c: In function ‘linuxnative_read_stream’:
format_linux_int.c:297:25: error: ‘SIOCGSTAMP’ undeclared (first use in this function); did you mean ‘SIOCGARP’?
   if (ioctl(stream->fd, SIOCGSTAMP,&tv)==0) {
                         ^~~~~~~~~~
                         SIOCGARP
format_linux_int.c:297:25: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [Makefile:731: format_linux_int.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
```

**Comment by salcock** [comment]
I believe cd7f4c79aa55823d2e3be9b753088c40bc44d183 should fix this issue.

The commit has now been merged into the develop branch -- could you try patching that fix in and seeing if it resolves the problem for you?

**Comment by jer-gentoo** [comment]
Yes, it does.

**Comment by salcock** [comment]
Great! I'll close this issue then and push the fix out in the next release.

Thanks for bringing this to my attention.

**Comment by talhahussain09** [comment]
Hi everyone i'm very new to linux and while installing BlueZ on my RaspberryPi I encountered this problem
Can you nice folks please tell me how to implement this solution

**Comment by salcock** [comment]
Hello,

You will need to reach out to the Bluez developers and ask them about the problem, as it'll be something that they need to fix in their code.

Feel free to link them to this issue page though -- the solution we applied to our code is probably the one that they need as well.

**Comment by jaui** [comment]
adding
`#include <linux/sockios.h>`
to the include section of BlueZ 
tools/l2test.c and tools/rctest.c 
solved the issue.

**Comment by 0bliss** [comment]
@jaui Thank you so much! Saved me.

**Comment by calvaradocl** [comment]
@jaui thanks for the heads up, helped us on the installation of BlueZ for Raspbian (bluez-5.9)

**Comment by hatem912000** [comment]
@jaui thanks for the heads up, helped us on the installation of BlueZ for Raspbian (bluez-5.8)

**Comment by EndlessRecurrence** [comment]
> adding `#include <linux/sockios.h>` to the include section of BlueZ tools/l2test.c and tools/rctest.c solved the issue.

Dude this is amazing, thank you!

**Comment by rajeevkr945** [comment]
> adding `#include <linux/sockios.h>` to the include section of BlueZ tools/l2test.c and tools/rctest.c solved the issue.

@jaui you are a lifesaver man. thankyou.

**Comment by dimcu** [comment]
@jaui Dude this is amazing, thank you!
